pkg/cmd/issue/comment/comment.go

```diff
@@ -0,0 +1,199 @@
+package comment
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/issue/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/surveyext"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type CommentOptions struct {
+	HttpClient          func() (*http.Client, error)
+	IO                  *iostreams.IOStreams
+	BaseRepo            func() (ghrepo.Interface, error)
+	EditSurvey          func() (string, error)
+	InputTypeSurvey     func() (int, error)
+	ConfirmSubmitSurvey func() (bool, error)
+	OpenInBrowser       func(string) error
+
+	SelectorArg string
+	Interactive bool
+	InputType   int
+	Body        string
+}
+
+const (
+	editor = iota
+	web
+	inline
```

Comment:
These consts could be made slightly more descriptive, e.g. `inputTypeEditor` instead of `editor`.